sentry-core/src/main/java/io/sentry/core/Breadcrumb.java

```diff
@@ -66,4 +69,50 @@ public void setLevel(SentryLevel level) {
   public void acceptUnknownProperties(Map<String, Object> unknown) {
     this.unknown = unknown;
   }
+
+  @TestOnly
+  Map<String, Object> getUnknown() {
+    return unknown;
+  }
+
+  @Override
+  public Breadcrumb clone() throws CloneNotSupportedException {
+    Breadcrumb clone = (Breadcrumb) super.clone();
+
+    clone.timestamp = timestamp != null ? (Date) timestamp.clone() : null;
+
+    if (data != null) {
+      Map<String, String> dataClone = new ConcurrentHashMap<>();
+
+      for (Map.Entry<String, String> item : data.entrySet()) {
+        if (item != null) {
+          dataClone.put(item.getKey(), item.getValue());
+        }
+      }
+
+      clone.data = dataClone;
+    } else {
+      clone.data = null;
+    }
+
+    if (unknown != null) {
+      Map<String, Object> unknownClone = new HashMap<>();
+
+      for (Map.Entry<String, Object> item : unknown.entrySet()) {
+        if (item != null) {
+          unknownClone.put(
+              item.getKey(),
+              item.getValue()); // TODO: how do we clone an object that we dont know the shape of it
```

Comment:
We can try to cast it to `Cloneable` and call clone though.

Otherwise just:
```suggestion
              item.getValue()); // NOTE: We shallow copy objects
```